libs/importer/src/importers/keeper/keeper-json-importer.ts

```diff
@@ -1,71 +1,558 @@
+        break;
+      default:
+        // Do nothing, default conversion is fine
+        break;
+    }
+
+    this.addField(cipher, importedName, importedValue, importedType);
+  }
+
+  private parseDate(timestamp: string | number): string {
+    const date = new Date(timestamp);
+    return isNaN(date.getTime()) ? "" : date.toLocaleString();
+  }
+
+  // This function parses custom field keys of the form:
+  // $<type>:<name>:<suffix> and returns [type, name]
+  // It handles a bunch of edge cases as well. See tests for examples.
+  // This function is modeled after the original implementation.
+  private parseFieldKey(key: string): [string, string] {
+    if (this.isNullOrWhitespace(key)) {
+      return ["", ""];
+    }
+
+    let fieldType = "";
+    let fieldName = "";
+
+    if (key[0] === "$") {
+      const pos = key.indexOf(":");
+      if (pos > 0) {
+        fieldType = key.substring(1, pos).trim();
+        fieldName = key.substring(pos + 1).trim();
+      } else {
+        fieldType = key.substring(1).trim();
+        fieldName = "";
+      }
+    } else {
+      fieldType = "";
+      fieldName = key;
+    }
+
+    if (
+      fieldName.length >= 2 &&
+      fieldName[fieldName.length - 2] === ":" &&
+      /\d/.test(fieldName[fieldName.length - 1])
+    ) {
+      fieldName = fieldName.substring(0, fieldName.length - 2);
+    }
+
+    return [fieldType, fieldName];
+  }
+
+  private addField(cipher: CipherView, name: string, value: any, type: FieldType = FieldType.Text) {
+    if (!value) {
+      return;
+    }
+
+    const field = new FieldView();
+    field.type = type;
+    field.name = name;
+    field.value = this.convertToFieldValue(value);
+    cipher.fields.push(field);
+  }
+
+  // Just to be safe, value come in all kinds of flavors
+  private convertToFieldValue(value: any): string {
+    if (typeof value === "string") {
+      return value;
+    }
+
+    try {
+      return JSON.stringify(value);
+    } catch {
+      // Fallthrough
+    }
+
+    return "";
+  }
+
+  private makeArray(value: any): any[] {
+    if (Array.isArray(value)) {
+      return value;
+    }
+    if (value != null) {
+      return [value];
+    }
+    return [];
+  }
+
+  private parseFolders(result: ImportResult, record: Record) {
+    if (!record.folders) {
       return;
     }
 
     record.folders.forEach((item) => {
       if (item.folder != null) {
-        this.processFolder(result, item.folder);
-        return;
+        this.processFolder(result, this.sanitizeFolderName(item.folder));
       }
 
       if (item.shared_folder != null) {
-        this.processFolder(result, item.shared_folder);
-        return;
+        this.processFolder(result, this.sanitizeFolderName(item.shared_folder));
       }
     });
   }
+
+  private sanitizeFolderName(name: string): string {
+    // `\` and `/` are reserved characters in Bitwarden, but valid in Keeper, replace them with `-`.
+    return name.replaceAll("\\\\", "-").replaceAll("/", "-");
```

Comment:
I think we're escaping `\` twice and only matches on `\\`?
```suggestion
    return name.replaceAll("\\", "-").replaceAll("/", "-");
```